lib/nostrum/struct/channel.ex

```diff
@@ -0,0 +1,266 @@
+        }
+
+  @type dm_channel :: %__MODULE__{
+          id: id,
+          type: 1,
+          guild_id: nil,
+          position: nil,
+          permission_overwrites: nil,
+          name: nil,
+          topic: nil,
+          nsfw: nil,
+          last_message_id: last_message_id,
+          bitrate: nil,
+          user_limit: nil,
+          recipients: recipients,
+          icon: nil,
+          owner_id: nil,
+          application_id: nil,
+          parent_id: nil,
+          last_pin_timestamp: nil
+        }
+
+  @type guild_voice_channel :: %__MODULE__{
+          id: id,
+          type: 2,
+          guild_id: guild_id,
+          position: position,
+          permission_overwrites: permission_overwrites,
+          name: name,
+          topic: nil,
+          nsfw: nsfw,
+          last_message_id: nil,
+          bitrate: bitrate,
+          user_limit: user_limit,
+          recipients: nil,
+          icon: nil,
+          owner_id: nil,
+          application_id: nil,
+          parent_id: parent_id,
+          last_pin_timestamp: nil
+        }
+
+  @type group_dm_channel :: %__MODULE__{
+          id: id,
+          type: 3,
+          guild_id: nil,
+          position: nil,
+          permission_overwrites: nil,
+          name: name,
+          topic: nil,
+          nsfw: nil,
+          last_message_id: last_message_id,
+          bitrate: nil,
+          user_limit: nil,
+          recipients: recipients,
+          icon: icon,
+          owner_id: owner_id,
+          application_id: application_id,
+          parent_id: nil,
+          last_pin_timestamp: nil
+        }
+
+  @type channel_category :: %__MODULE__{
+          id: id,
+          type: 4,
+          guild_id: guild_id,
+          position: position,
+          permission_overwrites: permission_overwrites,
+          name: name,
+          topic: nil,
+          nsfw: nsfw,
+          last_message_id: nil,
+          bitrate: nil,
+          user_limit: nil,
+          recipients: nil,
+          icon: nil,
+          owner_id: nil,
+          application_id: nil,
+          parent_id: parent_id,
+          last_pin_timestamp: nil
+        }
+
+  @type guild_channel ::
+          guild_text_channel
+          | guild_voice_channel
+          | channel_category
+
+  @type text_channel ::
+          guild_text_channel
+          | dm_channel
+          | group_dm_channel
+
+  @type voice_channel :: guild_voice_channel
+
+  @type t ::
+          guild_text_channel
+          | dm_channel
+          | guild_voice_channel
+          | group_dm_channel
+          | channel_category
+
+  @doc ~S"""
+  Formats a `Nostrum.Struct.Channel` into a mention.
+
+  ## Examples
+
+  ```Elixir
+  iex> channel = %Nostrum.Struct.Channel{id: 381889573426429952}
+  ...> Nostrum.Struct.Channel.mention(channel)
+  "<#381889573426429952>"
+  ```
+  """
+  @spec mention(t) :: String.t()
+  def mention(%__MODULE__{id: id}), do: "<##{id}>"
+
+  @doc false
+  def p_encode do
```

Comment:
We shouldn't need this function anymore, but if you want to leave it for now that's fine as that's what we've done in other modules.

Reply:
I've been keeping the `p_encode/0` functions in for compatibility with structs that haven't been updated. I do plan on removing these once we remove all `p_encode/0` references.

Reply:
Yep, that's fine.